icicle-data/src/Icicle/Common/Exp/Simp/Beta.hs

```diff
@@ -15,16 +15,16 @@ import P
 import Data.Hashable (Hashable)
 
 -- | Beta and let reduction
-beta :: (Hashable n, Eq n) => (Exp a n p -> Bool) -> Exp a n p -> Exp a n p
-beta isValue toplevel
+beta :: (Hashable n, Eq n) => Exp a n p -> Exp a n p
+beta toplevel
  = go toplevel
  where
   go xx
    = case xx of
       XApp _ p q
        | XLam _ n _ x <- go p
        , v <- go q
-       , isValue v
+       , isSimpleValue v
        , Just x' <- substMaybe n v x
        -> go x'
```

Comment:
it's if `a` is a value, not `y`. because the substitution might duplicate `a`, which isn't a good idea if `a` is expensive